lib/cli/ui/prompt/interactive_options.rb

```diff
@@ -86,15 +86,30 @@ def clear_output
 
         def num_lines
           options = presented_options.map(&:first)
-          # @options will be an array of questions but each option can be multi-line
-          # so to get the # of lines, you need to join then split
 
           # empty_option_count is needed since empty option titles are omitted
           # from the line count when reject(&:empty?) is called
-
           empty_option_count = options.count(&:empty?)
-          joined_options = options.join("\n")
-          joined_options.split("\n").reject(&:empty?).size + empty_option_count
+
+          # options will be an array of questions but each option can be multi-line
+          # so to get the # of lines, you need to join then split
+
+          # since lines may be longer than the terminal is wide, we need to
+          # determine how many extra lines would be taken up by them
+          max_width = (CLI::UI::Terminal.width -
+                       options.count.to_s.size - # Width of the displayed number
+                       5 -                       # Extra characters added during rendering
```

Comment:
I'm not a fan of having this magic number in here, but I couldn't figure out a reasonable way to account for the extra characters added during rendering without it.